### PR TITLE
refactor: swap from difference to similar crate

### PR DESCRIPTION
**Pull request by 20jasper**
`difference` has not been updated for 8 years, so swap to `similar` crate. https://rustsec.org/advisories/RUSTSEC-2020-0095.html

Also fixes an off by one error. Trailing newlines were counted as a line

### CODE REVIEW

File: tests/snapshots/success-include-directive-in-new-project-test-added-no-dryrun [hunk 1]
```diff
@@ -14,4 +14,4 @@ edition = "2018"
 +include = ["src/**/*", "README.md", "!**/*_test.*"]
 
 [dependencies]
-[…skipped 1 lines…]
\ No newline at end of file
+[…skipped 0 lines…]
```
Comment (20jasper) [inline tests/snapshots/success-include-directive-in-new-project-test-added-no-dryrun]: Previously this counted the newline at the end of the file, which seems like an off by one error to me

Comment (Byron) [reply tests/snapshots/success-include-directive-in-new-project-test-added-no-dryrun]: Now it seems the `[…skipped 0 lines…]` has no use anymore.
Removing it would also be a good test for the suspected off-by-one error - will it still look right?

Comment (20jasper) [reply tests/snapshots/success-include-directive-in-new-project-test-added-no-dryrun]: > Now it seems the […skipped 0 lines…] has no use anymore.

Good callout, updated to remove the directive and added a new trailing truncation test

**Comment by Byron** [comment]
This is great, thanks so much.

I will also create a new release shortly.
